workload/ttllogger: introduce example TTL workload

otan opened the pull request:
Release justification: high priority introduction for demostrating new
functionality

Release note (cli change): Introduce a new `ttllogger` workload which
creates a TTL table emulating a "log" with rows expiring after the
duration specified in the `--ttl` flag.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/77649)
<!-- Reviewable:end -->

otan committed:
workload/ttllogger: introduce example TTL workload

Release justification: high priority introduction for demostrating new
functionality

Release note (cli change): Introduce a new `ttllogger` workload which
creates a TTL table emulating a "log" with rows expiring after the
duration specified in the `--ttl` flag.

rafiss reviewed (approved):
whoops i missed this! lgtm as is!

 would we want a roachtest from this too?

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/77649#-:-Mzvgfc50q3la8sExnYl:bssgytv)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
> would we want a roachtest from this too?

i guess we could, but not sure i see the value in it (yet)

for now though...

bors r=rafiss
